packages/ozone/src/api/admin/updateCommunicationTemplate.ts

```diff
@@ -0,0 +1,36 @@
+import { AuthRequiredError, InvalidRequestError } from '@atproto/xrpc-server'
+import { Server } from '../../lexicon'
+import AppContext from '../../context'
+
+export default function (server: Server, ctx: AppContext) {
+  server.com.atproto.admin.updateCommunicationTemplate({
+    auth: ctx.roleVerifier,
+    handler: async ({ input, auth }) => {
+      const access = auth.credentials
+      const db = ctx.db
+      const { id, updatedBy, ...template } = input.body
+
+      if (!access.moderator) {
+        throw new AuthRequiredError(
+          'Must be a full moderator to update a communication template',
+        )
+      }
+
+      if (!Object.keys(template).length) {
+        throw new InvalidRequestError('Missing update data in request body')
+      }
+
+      const communicationTemplate = ctx.communicationTemplateService(db)
+      const updatedTemplate = await communicationTemplate.update(id, {
+        ...template,
+        updatedAt: new Date(),
```

Comment:
It's small peas, but it might be nice to maintain the `updatedAt` and `createdAt` times as logic within the template service rather than have any/all callers deal with it.  It's not uncommon for it to be set even lower in the stack, e.g. setting `createdAt` via a default to `now()` within the database itself.

Reply:
makes sense, I see the default being set at the service layer in other places so going to use that approach.